resources/rewrite_handler_production.py

```diff
@@ -11,18 +12,29 @@ def handler(event, context):
     request = event['Records'][0]['cf']['request']
     params = {k : v[0] for k, v in parse_qs(request['querystring']).items()}
 
-    if ('id' in params):
+    if ('id' in params and 'url' not in params):
             s3_root = "s3://figgy-geo-production"
 
             if ('mosaicjson' in request['uri']):
-                file_name = 'mosaic.json'
+                # Strategy - if it's Mosaic URL, then fetch S3 URL from
+                # https://figgy.princeton.edu/concern/raster_resources/<id>/mosaic.json,
+                # parse JSON and get URI parameter.
+                item_id = params['id']
+                item_url = figgy_s3_url(item_id)
             else:
                 file_name = 'display_raster.tif'
+                item_id = params['id']
+                item_url = f"{s3_root}/{item_id[0:2]}/{item_id[2:4]}/{item_id[4:6]}/{item_id}/{file_name}"
 
-            item_id = params['id']
-            item_url = f"{s3_root}/{item_id[0:2]}/{item_id[2:4]}/{item_id[4:6]}/{item_id}/{file_name}"
 
             params['url'] = item_url
             request['querystring'] = urlencode(params)
 
     return request
+
+def figgy_uri(figgy_id):
+    return f"https://figgy.princeton.edu/concern/raster_resources/{figgy_id}/mosaic.json"
+
+def figgy_s3_url(figgy_id):
+    resp = requests.get(url=figgy_uri(figgy_id))
```

Comment:
question: Should we add the fallback to the non-fingerprinted mosaic.json in this PR? 

- when status code isn't 200
- catching an exception in a try block